### ISSUE TITLE
Bandwidth Monitoring not working on Debian 12?

### ISSUE THREAD

**Issue by Sebastian1989101** [webmin/webmin#2307]
I have a fresh installed Debian 12 vServer and just added Webmin. One reason I added Webmin was for the Bandwidth Monitoring as this Server works as a VPN. However, the module just reports: 

`Neither of the System Logs modules are installed on this system and supported by your OS. The Bandwidth Monitoring module cannot operate without one of them.`

The command `journalctl -k --grep "BANDWIDTH_IN|BANDWIDTH_OUT"` also just yields `-- No entries --`

**Comment by iliajie** [comment]
Hey, Sebastian!

Thanks for the heads up! I like your suggestion! You’re absolutely right—we need to fix this! The problem is that most modern systems no longer use `rsyslog` and rely on `firewalld` to interact with `iptables`.

I dug in a bit and found that it might be simpler than I initially thought.

Try running this bash script, replacing the interface name (and possibly zone name) with the one you’re using:

```bash
#!/usr/bin/env bash

# Incoming traffic logging without rate limit (limit could be imposed using `limit value="1/m"`)
firewall-cmd --permanent --zone=public --add-rich-rule='rule family="ipv4" source address="0.0.0.0/0" log prefix="BANDWIDTH_IN: " level="info" accept'

# Outgoing traffic logging without rate limit
firewall-cmd --permanent --zone=public --add-rich-rule='rule family="ipv4" destination address="0.0.0.0/0" log prefix="BANDWIDTH_OUT: " level="info" accept'

# Reload FirewallD to apply changes
firewall-cmd --reload

# Verify the rules are in place
firewall-cmd --zone=public --list-rich-rules
```

Then you'd be able to get bandwidth logs using:

```bash
journalctl -f -t kernel | grep "BANDWIDTH_"
```

**Comment by iliajie** [comment]
@jcameron Jamie, what are your thoughts on this? Should we fix bandwidth monitoring for systems using `firewalld` and `journalctl`?

**Comment by jcameron** [comment]
@iliajie yes that sounds like a good idea, although potentially complex to support firewalld since we'd first need to add support for rich rules in Webmin.

**Comment by iliajie** [comment]
I don’t think we need to care about rich rules support at all, as it’s handled by FirewallD, right?

**Comment by iliajie** [comment]
Can you elaborate on this? This is handled by `firewall-cmd` command? What do you mean by saying “we do need to support that”?

**Comment by iliajie** [comment]
Let’s clear this up—I’d like to get it working...

**Comment by jcameron** [comment]
What I mean is, the current firewalld module doesn't support that flag in it's API. We could certainly add this, but it would be a project ...

**Comment by iliajie** [comment]
> What I mean is, the current firewalld module doesn't support that flag in it's API

What flag are you referring to!?

I think all we need to do is just run the following:

```
# Incoming traffic logging without rate limit (limit could be imposed using `limit value="1/m"`)
firewall-cmd --permanent --zone=public --add-rich-rule='rule family="ipv4" source address="0.0.0.0/0" log prefix="BANDWIDTH_IN: " level="info" accept'

# Outgoing traffic logging without rate limit
firewall-cmd --permanent --zone=public --add-rich-rule='rule family="ipv4" destination address="0.0.0.0/0" log prefix="BANDWIDTH_OUT: " level="info" accept'
```

And then retrieve logs using something like:

```
journalctl -t kernel | grep "BANDWIDTH_"
```

**Comment by iliajie** [comment]
> > What I mean is, the current firewalld module doesn't support that flag in it's API
> 
> What flag are you referring to!?

If you mean the Webmin module, then we don’t need to add any specific API to handle this. We can simply run the command directly as we do in [other](https://github.com/webmin/webmin/commit/6b190a6cbae04b9e7d3ac579cca2f099a87e0c25#diff-f8cae0565bfd7237881d2e2456a9c7efa1633b9e47ade11ed67098a977002d5fR515-R518) cases.

**Comment by jcameron** [comment]
Better to do it via the webmin API though. However, I was wrong ... we do have support for rich rules already : https://github.com/webmin/webmin/blob/master/firewalld/firewalld-lib.pl#L405

**Comment by iliajie** [comment]
> However, I was wrong ... we do have support for rich rules already

Oh, I was wrong too—I totally forgot that we added it a couple of years ago.

Then, I’ll look into fixing bandwidth monitoring, and send you a PR later for the review.

**Comment by iliajie** [comment]
> Then, I’ll look into fixing bandwidth monitoring, and send you a PR later for the review.

I'm starting to look into it...

**Comment by iliajie** [comment]
Implemented here https://github.com/webmin/webmin/pull/2371.
